scripts/validate-index.ts

```diff
@@ -0,0 +1,25 @@
+/* Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import Ajv from 'ajv';
+import * as colours from 'ansi-colors';
+
+import orgIndexSchema from '../resources/org_index_schema.json';
+import { readOrgIndexFiles } from './helpers';
+
+async function run() {
+    const ajv = new Ajv();
+    const files = await readOrgIndexFiles();
+
+    for (const { id, orgIndex } of files) {
+        if (!ajv.validate(orgIndexSchema, orgIndex)) {
+            throw new Error(`Index file "${id}" does not match the schema.`);
+        }
```

Comment:
We need to provide some useful output here.
`ajv` provides error objects if we use `compile` to generate the validator:

```suggestion
    const validate = ajv.compile(orgIndexSchema);
    for (const { id, file, orgIndex } of files) {
        const isValid = validate(orgIndex);
        if (!isValid) {
            validate.errors.forEach(error => reportError(file, error));
            throw new Error(`Index file "${id}" does not match the schema.`);
        }
```

`reportError` isn't defined here, but should tie into github's [error reporting mechanism](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-error-message). For instance:
```ts
function reportError(file: string, error: ErrorObject) {
    const message = `Property ${error.instancePath} ${error.message}`;
    if ('GITHUB_RUN_ID' in process.env) {
        console.log(`::error file={${file}},title={Invalid index file}::${message}`);
    } else {
        console.error(colours.red(`Error in ${file}: ${message}`));
    }
}
```
This could potentially be generalized into a generic error function and used in other files.

I'd also argue that throwing errors doesn't produce particularly readable error output, and since it's happening in the middle of the for loop, we'd end up only seeing one error at the time. When getting error output from CI, I'd like to see all the problems right away, so I can fix them all, then push, instead of fixing one, pushing, then seeing the next one, then pushing. We could for example just count the number of errors, and pass that to `process.exit()` at the end to let GitHub know whether everything worked without bailing early.

Reply:
> When getting error output from CI, I'd like to see all the problems right away, so I can fix them all, then push, instead of fixing one, pushing, then seeing the next one, then pushing

In other contexts I'd agree, but in the context in which this will be run I don't think that makes sense. This will be run when organisations make PRs contributing their JSON schema, so there's only the possibility for one error at a time (assuming all the others have been validated beforehand, which they would have been given the action hooks).

But tying into GitHub's reporting mechanism definitely makes sense 👍

Reply:
> assuming all the others have been validated beforehand, which they would have been given the action hooks

I don't think this is a valid reason for hiding errors when there's no consequence from showing all of them. CI should always show all independent errors on principle, regardless of assumptions